Spigot-Server-Patches/0418-Ticket-level-per-player-view-distance-API.patch

```diff
@@ -0,0 +1,186 @@
+From a1c6af9725915f855ab2cf02e22281ece25cac87 Mon Sep 17 00:00:00 2001
+From: slicklibro <slicklibro@gmail.com>
+Date: Fri, 20 Sep 2019 17:16:14 +1200
+Subject: [PATCH] Ticket level per-player view distance API
+
+This PR implements the per-player view distance API at the ticket level.
+This somewhat clean solution comes from the result of many complicated ones (and research).
+The view-distance config value will determine the global distance (and set maximum) of normal view distance interactions with entities, mob-spawning and tracking.
+All chunks with player tickets within this range will also be served by the server, allowing shared view distance from other players without impacting performance.
+
+diff --git a/src/main/java/net/minecraft/server/ChunkMapDistance.java b/src/main/java/net/minecraft/server/ChunkMapDistance.java
+index 63a688725..879099b20 100644
+--- a/src/main/java/net/minecraft/server/ChunkMapDistance.java
++++ b/src/main/java/net/minecraft/server/ChunkMapDistance.java
+@@ -43,6 +43,7 @@ public abstract class ChunkMapDistance {
+     private final LongSet l = new LongOpenHashSet();
+     private final Executor m;
+     private long currentTick;
++    protected ObjectSet<EntityPlayer> getChunkPlayers(long i) {return (ObjectSet) ChunkMapDistance.this.c.get(i);} //Paper OBFHELPER
```

Comment:
this is not an obfhelper, this actually requires an obfhelper on c

Reply:
noted and will be pushed on next commit